src/Symfony/Component/Notifier/Bridge/Gitter/GitterTransport.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Gitter;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Christin Gruber <c.gruber@touchdesign.de>
+ */
+final class GitterTransport extends AbstractTransport
+{
+    protected const HOST = 'api.gitter.im';
+
+    private $token;
+    private $roomId;
+
+    public function __construct(string $token, string $roomId, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->token = $token;
+        $this->roomId = $roomId;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('gitter://%s?roomId=%s', $this->getEndpoint(), $this->roomId);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage;
+    }
+
+    /**
+     * @see https://developer.gitter.im/docs/rest-api
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $options = ($opts = $message->getOptions()) ? $opts->toArray() : [];
```

Comment:
Looks likt this bridge does not have dedicated options, so this line could be removed and $options can be set directly in the `json` key down below

Reply:
So true, done.